### PR TITLE
feat(cast): reuse nonce

### PR DESCRIPTION
**Pull request by calldata**
reuse nonce

### CODE REVIEW

File: cli/src/cast.rs [hunk 1]
```diff
@@ -232,17 +232,30 @@ async fn main() -> eyre::Result<()> {
...
+                    WalletType::Ledger(leger) => leger.address(),
+                    WalletType::Local(local) => local.address(),
+                    WalletType::Trezor(trezor) => trezor.address(),
+                };
+
+                let nonce = if resend {
+                    Some(provider.get_transaction_count(from, None).await?)
+                } else {
+                    nonce
+                };
```
Comment (mattsse) [inline cli/src/cast.rs:252]: make `mut nonce` and then 
`nonce = Some(provider.get_transaction_count(from, None).await?);`

File: cli/src/cast.rs [hunk 2]
```diff
@@ -298,6 +311,11 @@ async fn main() -> eyre::Result<()> {
                     }
                 }
             } else if let Some(from) = eth.from {
+                let nonce = if resend {
+                    Some(provider.get_transaction_count(from, None).await?)
+                } else {
+                    nonce
+                };
```
Comment (mattsse) [inline cli/src/cast.rs:318]: same

**Review by mattsse — CHANGES_REQUESTED** [review]

File: cli/src/opts/cast.rs [hunk 3]
```diff
@@ -254,6 +254,8 @@ pub enum Subcommands {
         confirmations: usize,
         #[clap(long = "json", short = 'j')]
         to_json: bool,
+        #[clap(long = "resend", help = "reuse nonce")]
+        resend: bool,
```
Comment (mattsse) [inline cli/src/opts/cast.rs:258]: this should be conflicts_with = nonce, right?
can we also be a bit more explicit in the help message?

**Review by gakonst — APPROVED** [review]
LGTM - test is a flake

**Pull request merged**